app/components/markdown/markdown_table_image/markdown_table_image.js

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {PropTypes} from 'prop-types';
+import React from 'react';
+import {intlShape} from 'react-intl';
+import {Text} from 'react-native';
+
+import CustomPropTypes from 'app/constants/custom_prop_types';
+import {wrapWithPreventDoubleTap} from 'app/utils/tap';
+
+export default class MarkdownTableImage extends React.PureComponent {
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+        source: PropTypes.string.isRequired,
+        textStyle: CustomPropTypes.Style.isRequired,
+        navigator: PropTypes.object.isRequired,
+        serverURL: PropTypes.string.isRequired,
+        theme: PropTypes.object.isRequired
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired
+    };
+
+    handlePress = wrapWithPreventDoubleTap(() => {
+        const {navigator, theme} = this.props;
+
+        navigator.push({
+            screen: 'TableImage',
```

Comment:
Wondering if we should use a [LightBox](https://wix.github.io/react-native-navigation/#/screen-api?id=showlightboxparams-) instead

Reply:
I'll take a look at doing that. I was actually wanting to use the same image preview as we have for file attachments, but I couldn't figure out how to decouple that from the file attachments

Reply:
Alright, well in the end I think the Image preview needs to be refactored anyway